joomla/joomla-cms
# [4.0] Clean up MvcFactory

Pull Request for Issue # .

### Summary of Changes
The current MVC factory implement completely ignore $prefix parameter. That result the code in backend cannot create MVC objects from frontend and vice versa.  There are also several small clean up:

1. Force table class to use Administrator namespace unless Site is directly passed
2. Use correct CmsApplication class instead of it's alias JApplicationCms

### Testing Instructions
Maybe code review would be enough. Or install Joomla 4 nightly, apply patch, play with com_content and make sure it still works as expected
